tutorials/nemotron-cc/nemotron_cc.ipynb

```diff
@@ -0,0 +1,4227 @@
+   "source": [
+    "# Building Nemotron-CC data curation pipeline using Nemo Curator\n",
+    "\n",
+    "This tutorial demonstrates how to build Nemotron-CC data curation pipeline (https://arxiv.org/pdf/2412.02595) using NeMo Curator. The NeMo Curator is a Python library that consists of a collection of scalable data-mining modules for curating natural language processing (NLP) data for training large language models (LLMs). It consists of easy to use modules for data download, extraction, language identification, quality filtering, deduplication to build high-quality data curation pipelines at scale from massive uncurated web corpora.\n",
+    "\n",
+    "## Nemotron-cc curation pipeline components\n",
+    "\n",
+    "- Common Crawl data download and extraction\n",
+    "- Language Identification and filtering\n",
+    "- Exact and fuzzy deduplication\n",
+    "- Heuristic filtering and Perplexity filtering\n",
+    "- Model based quality labeling\n",
+    "- Postprocessing"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b9c41cfd",
+   "metadata": {},
+   "source": [
+    "## Flowchart"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3f5c4865",
+   "metadata": {},
+   "source": [
+    "# <img src=\"images/nemotron_cc_diagram.png\" alt=\"Nemotron-CC Diagram\" width=\"800\"/>\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ebe32441",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Display each individual phases in the flow chart\n",
+    "# !pip install mermaid-python\n",
+    "from mermaid import Mermaid\n",
+    "\n",
+    "with open(\"viz/flowchart.txt\", \"r\") as f:\n",
+    "    mermaid_code = f.read()\n",
+    "\n",
+    "# Create and display the diagram\n",
+    "Mermaid(mermaid_code)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "78537bd7",
+   "metadata": {},
+   "source": [
+    "## Prerequisites\n",
+    "\n",
+    "### System Requirements\n",
+    "Here is the hardware setting for this notebook\n",
+    "\n",
+    "**GPU**: NVIDIA A100 80GiB.\n",
+    "\n",
+    "**CUDA & Nvidia Drivers**: CUDA 12.2 with Driver 535.154.05\n",
+    "\n",
+    "**OS**: ubuntu 22.04\n",
+    "\n",
+    "### Getting NeMo Framework Training Container\n",
+    "- Get access to the container via https://developer.nvidia.com/nemo-framework\n",
+    "- Set your docker credentials \n",
+    "    ```bash\n",
+    "    docker login nvcr.io\n",
+    "\n",
+    "    Username: $oauthtoken\n",
+    "    Password: <Your NGC Key>\n",
+    "- Get NeMo NeMo Framework Training Container\n",
+    "    ```bash\n",
+    "    docker pull nvcr.io/nvidia/nemo:25.05.rc2\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "062b5423",
+   "metadata": {},
+   "source": [
+    "## 0. Env Setup"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "60d6bc26",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# !pip install -r requirements.txt"
```

Comment:
Delete this cell, unless there are additional requirements not already provided in the container?